extensions/amp-date-display/0.1/test/test-amp-date-display.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../amp-date-display';
+
+describes.realWin('amp-date-display', {
+  amp: {
+    extensions: ['amp-date-display'],
+  },
+}, env => {
+
+  let win;
+  let element;
+  let impl;
+
+  beforeEach(() => {
+    win = env.win;
+    element = win.document.createElement('amp-date-display');
+    win.document.body.appendChild(element);
+    impl = element.implementation_;
```

Comment:
I copied this from `amp-date-countdown`:

https://github.com/ampproject/amphtml/blob/08d876d53ccc10dfe8e44671c71b64fce48b4857/extensions/amp-date-countdown/0.1/test/test-amp-date-countdown.js#L54

Is it really wrong? The tests pass.

Also the suggested change does not work since the promise is never settled and so `beforeEach` times out. I could use `element.implementation_` directly in each test.

Reply:
Ah ok, thanks for checking. I made an assumption based on my knowledge of the API. If these tests become flaky later this could be something good to look at.